### DISCUSSION TITLE
Nomad service discovery

### DISCUSSION THREAD

**Discussion by m1keil** [prometheus/prometheus#10820] [Ideas]
I'm interested in creating a PR for Nomad SD. Before I go ahead and spend some time on it, I wonder if such a PR would be endorsed by Prometheus?

Why not Consul SD?
Yes, Nomad is typically coupled with Consul to provide service registration mechanism. However, latest Nomad release already introduced a "Consul-less" service discovery mechanism. So it would natural for Nomad to have its own "native" SD. 

Would be curious to hear your opinion. 
Cheers.

**Comment by roidelapluie** [comment]
How stable is this upstream? Is it also available when using consul?

Le sam. 4 juin 2022, 12:06, Michael Sverdlik ***@***.***> a
écrit :

**Comment by SamMousa** [comment]
I'd love this! 
I couldn't find clearly in the consul sd documentation, but it would also be nice to identify exporter services running inside the cluster.
For example if I have 3 beanstalkd nodes running in nomad. I'll put an exporter with each one which exposes a service with a tag, say `prometheus-path=/metrics`. 
The service discovery should then look at all services with this tag and proceed to scrape all endpoints with the given path.

**Reply by roidelapluie** [reply]
This can be done with relabeling. And Nomad SD has been implemented, if you are not using consul

**Reply by tonnyavery** [reply]
I wish there would be more examples and explanation on how to do relabling for nomad service discovery.
With consul it is relatively straightforward, you can provide services and tags. Nomad service discovery implementation doesn't support that.
